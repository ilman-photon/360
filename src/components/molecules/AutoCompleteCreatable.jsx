import * as React from "react";
import Autocomplete, { createFilterOptions } from "@mui/material/Autocomplete";
import StyledInput from "../atoms/Input/input";
import { CircularProgress, Box, Typography } from "@mui/material";
import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
import SearchIcon from "@mui/icons-material/Search";

const filter = createFilterOptions();

function sleep(delay = 0) {
  return new Promise((resolve) => {
    setTimeout(resolve, delay);
  });
}

export const AutoCompleteCreatable = ({
  isLoading = false,
  options = [],
  onFetch = () => {
    // This is intended
  },
  onInputEmpty = () => {
    // This is intended
  },
  ...props
}) => {
  const [open, setOpen] = React.useState(false);
  const loading = open && options.length === 0 && isLoading;
  const borderColor = props.error ? "#B00020" : "#BDBDBD";
  let backgroundColor = props.error ? "#FBF4F4" : "white";

  React.useEffect(() => {
    let active = true;

    if (!loading) {
      return undefined;
    }

    (async () => {
      await sleep(1e3); // For demo purposes.

      if (active) {
        onFetch();
      }
    })();

    return () => {
      active = false;
    };
  }, [loading, onFetch]);

  React.useEffect(() => {
    if (!open) {
      onInputEmpty();
    }
  }, [open, onInputEmpty]);

  return (
    <>
      <Autocomplete
        popupIcon={<KeyboardArrowDownIcon />}
        data-testid={props.testId}
        open={open}
        onOpen={() => {
          setOpen(true);
        }}
        onClose={() => {
          setOpen(false);
        }}
        onFocus={() => (backgroundColor = "white")}
        value={props.value}
        //   onChange={props.onChange}
        onChange={(_e, newValue) => {
          if (typeof newValue === "string") {
            props.onChange({
              label: newValue,
            });
          } else if (newValue && newValue.value) {
            // Create a new value from the user input
            props.onChange(newValue);
          } else {
            props.onChange(newValue);
          }
        }}
        isOptionEqualToValue={(option, value) => {
          if (value.label) {
            return option.label === value.label;
          } else if (value) {
            return option.label === value;
          } else {
            return false;
          }
        }}
        filterOptions={(optionsContext, params) => {
          const filtered = filter(optionsContext, params);
          const { inputValue } = params;
          // Suggest the creation of a new value
          const isExisting = optionsContext.some(
            (option) => inputValue === option.label
          );
          if (inputValue !== "" && !isExisting) {
            filtered.push({
              id: Math.floor(Math.random() * 100000),
              label: `Add "${inputValue}"`,
              value: inputValue,
            });
          }

          return filtered;
        }}
        selectOnFocus
        clearOnBlur
        handleHomeEndKeys
        options={options}
        getOptionLabel={(option) => {
          // Value selected with enter, right from the input
          if (typeof option === "string") {
            return option;
          }
          // Add "xxx" option created dynamically
          if (option.value) {
            return option.value;
          }
          // Regular option
          return option.label;
        }}
        loading={loading}
        ListboxProps={{
          sx: {
            fontSize: 14,
          },
        }}
        //   renderOption={(props, option) => <li {...props}>{option.label}</li>}
        renderOption={(params, option) => {
          return (
            <li {...params} key={option.id}>
              {option.label}
            </li>
          );
        }}
        renderInput={(params) => {
          return (
            <Box
              sx={{
                display: "flex",
                alignItems: "center",
                border: `1px solid ${borderColor}`,
                borderRadius: 1,
                backgroundColor: { backgroundColor },
              }}
            >
              <SearchIcon sx={{ height: 22, width: 22, pl: 1 }} />
              <StyledInput
                type="text"
                variant="filled"
                {...params}
                label={props.inputLabel}
                //error={props.error}
                //helperText={props.helperText}
                InputProps={{
                  ...params.InputProps,
                  sx: {
                    height: 52,
                    "& .MuiFilledInput-input": {
                      font: "",
                    },
                  },
                  endAdornment: (
                    <React.Fragment>
                      {loading ? (
                        <CircularProgress color="inherit" size={20} />
                      ) : null}
                      {params.InputProps.endAdornment}
                    </React.Fragment>
                  ),
                }}
                sx={{
                  "& .MuiInputLabel-formControl": {
                    color: props.error ? "#B00020" : "rgba(0, 0, 0, 0.6)",
                  },
                  "& .MuiFilledInput-root": {
                    border: "0px",
                    backgroundColor: { backgroundColor },
                  },
                  "& .MuiFilledInput-input": {
                    font: "",
                  },
                  "&:hover": {
                    backgroundColor: "transparent",
                  },
                }}
              />
            </Box>
          );
        }}
      />
      {props.error && (
        <Typography
          variant={"lightError"}
          sx={{
            ml: 1.75,
            mt: 0.5,
          }}
        >
          {props.helperText}
        </Typography>
      )}
    </>
  );
};
