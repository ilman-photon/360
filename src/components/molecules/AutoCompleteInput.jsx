import * as React from "react";
import Autocomplete from "@mui/material/Autocomplete";
import CircularProgress from "@mui/material/CircularProgress";
import StyledInput from "../atoms/Input/input";

function sleep(delay = 0) {
  return new Promise((resolve) => {
    setTimeout(resolve, delay);
  });
}

export const AutoCompleteInput = ({
  options = [],
  onFetch = () => {
    // This is intended
  },
  onInputEmpty = () => {
    // This is intended
  },
  ...props
}) => {
  const [open, setOpen] = React.useState(false);
  const loading = open && options.length === 0;

  React.useEffect(() => {
    let active = true;

    if (!loading) {
      return undefined;
    }

    (async () => {
      await sleep(1e3); // For demo purposes.

      if (active) {
        onFetch();
      }
    })();

    return () => {
      active = false;
    };
  }, [loading, onFetch]);

  React.useEffect(() => {
    if (!open) {
      onInputEmpty();
    }
  }, [open]);

  return (
    <Autocomplete
      open={open}
      onOpen={() => {
        setOpen(true);
      }}
      onClose={() => {
        setOpen(false);
      }}
      defaultValue={props.defaultValue}
      isOptionEqualToValue={(option, value) => {
        console.log({ option, value });
        if (value.label) {
          return option.label === value.label;
        } else if (value) {
          return option.label === value;
        } else {
          return false;
        }
      }}
      getOptionLabel={(option) => (option.label ? option.label : "")}
      options={options}
      loading={loading}
      ListboxProps={{
        sx: {
          fontSize: 14,
        },
      }}
      renderOption={(params, option) => {
        return (
          <li {...params} key={option.id}>
            {option.label}
          </li>
        );
      }}
      renderInput={(params) => {
        return (
          <StyledInput
            type="text"
            {...params}
            label={props.inputLabel}
            error={props.error}
            helperText={props.helperText}
            InputProps={{
              ...params.InputProps,
              sx: { height: 52 },
              endAdornment: (
                <React.Fragment>
                  {loading ? (
                    <CircularProgress color="inherit" size={20} />
                  ) : null}
                  {params.InputProps.endAdornment}
                </React.Fragment>
              ),
            }}
          />
        );
      }}
      onChange={props.onChange}
      value={props.value}
    />
  );
};

export default AutoCompleteInput;
