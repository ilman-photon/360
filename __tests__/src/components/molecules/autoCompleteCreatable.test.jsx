import { render, fireEvent, within } from "@testing-library/react";

import "@testing-library/jest-dom";
import { AutoCompleteCreatable } from "../../../../src/components/molecules/AutoCompleteCreatable";

describe("Auto Complete Creatable Components", () => {
  let container;
  const mocFetch = jest.fn();

  beforeEach(() => {
    const options = [
      { id: "01", value: "test1" },
      { id: "02", value: "test2" },
      { id: "03", value: "test3" },
    ];
    container = render(
      <AutoCompleteCreatable
        isLoading={true}
        onFetch={mocFetch}
        options={options}
        testId={"autocomplete"}
        onChange={jest.fn}
      />
    );
  });

  it("Auto Complete Creatable test", async () => {
    const autocomplete = container.getByTestId("autocomplete");
    const input = within(autocomplete).getByRole("combobox");
    autocomplete.focus();
    fireEvent.change(input, { target: { value: "test1" } });
    fireEvent.keyDown(autocomplete, { key: "ArrowDown" });
    fireEvent.keyDown(autocomplete, { key: "Enter" });
  });

  it("Auto Complete Creatable with label test", async () => {
    const data = [
      { label: "The Shawshank Redemption", year: 1994 },
      { label: "The Godfather", year: 1972 },
    ];
    container = render(
      <AutoCompleteCreatable
        isLoading={true}
        onFetch={mocFetch}
        options={data}
        testId={"new-autocomplete"}
        onChange={jest.fn}
      />
    );
    const autocomplete = container.getByTestId("new-autocomplete");
    const input = within(autocomplete).getByRole("combobox");
    autocomplete.focus();
    fireEvent.change(input, { target: { value: "test1" } });
    fireEvent.keyDown(autocomplete, { key: "ArrowDown" });
    fireEvent.keyDown(autocomplete, { key: "Enter" });
  });


  it("Auto Complete Creatable with new label test", async () => {
    const data = ["test1", "test2"];
    container = render(
      <AutoCompleteCreatable
        isLoading={true}
        onFetch={mocFetch}
        options={data}
        testId={"new-autocomplete1"}
        onChange={jest.fn}
      />
    );
    const autocomplete = container.getByTestId("new-autocomplete1");
    const input = within(autocomplete).getByRole("combobox");
    autocomplete.focus();
    fireEvent.change(input, { target: { value: "test1" } });
    fireEvent.keyDown(autocomplete, { key: "ArrowDown" });
    fireEvent.keyDown(autocomplete, { key: "Enter" });
  });

  it("Auto Complete Loading true", async () => {
    const data = [];
    container = render(
      <AutoCompleteCreatable
        isLoading={true}
        onFetch={mocFetch}
        options={data}
        testId={"new-autocomplete"}
        onChange={jest.fn}
      />
    );
    const autocomplete = container.getByTestId("new-autocomplete");
    const input = within(autocomplete).getByRole("combobox");
    autocomplete.focus();
    fireEvent.change(input, { target: { value: "test1" } });
  });

   it("Auto Complete No Props", async () => {
     container = render(
       <AutoCompleteCreatable />
     );
     expect(container).toMatchSnapshot();
   });

   it("Auto Complete With Error", async () => {
     container = render(<AutoCompleteCreatable error={true} />);
     expect(container).toMatchSnapshot();
   });
});
