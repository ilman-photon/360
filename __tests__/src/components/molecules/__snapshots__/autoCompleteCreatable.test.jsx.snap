// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auto Complete Creatable Components Auto Complete No Props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root"
        data-testid="autocomplete"
      >
        <div
          class="MuiBox-root css-18okncf"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gobr0o-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root custom-input css-hzc2jn-MuiFormControl-root-MuiTextField-root"
            tabindex="0"
            withicon="true"
          >
            <div
              class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-dvqt9d-MuiInputBase-root-MuiFilledInput-root"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1gctnaj-MuiInputBase-input-MuiFilledInput-input"
                id=":re:"
                placeholder=""
                role="combobox"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        class="MuiTypography-root MuiTypography-lightError css-1o6qin5-MuiTypography-root"
      />
    </div>
    <div>
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root"
      >
        <div
          class="MuiBox-root css-18okncf"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gobr0o-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root custom-input css-hzc2jn-MuiFormControl-root-MuiTextField-root"
            tabindex="0"
            withicon="true"
          >
            <div
              class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-dvqt9d-MuiInputBase-root-MuiFilledInput-root"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1gctnaj-MuiInputBase-input-MuiFilledInput-input"
                id=":rg:"
                placeholder=""
                role="combobox"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        class="MuiTypography-root MuiTypography-lightError css-1o6qin5-MuiTypography-root"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root"
    >
      <div
        class="MuiBox-root css-18okncf"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gobr0o-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root custom-input css-hzc2jn-MuiFormControl-root-MuiTextField-root"
          tabindex="0"
          withicon="true"
        >
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-dvqt9d-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1gctnaj-MuiInputBase-input-MuiFilledInput-input"
              id=":rg:"
              placeholder=""
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      class="MuiTypography-root MuiTypography-lightError css-1o6qin5-MuiTypography-root"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Auto Complete Creatable Components Auto Complete With Error 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root"
        data-testid="autocomplete"
      >
        <div
          class="MuiBox-root css-18okncf"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gobr0o-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root custom-input css-hzc2jn-MuiFormControl-root-MuiTextField-root"
            tabindex="0"
            withicon="true"
          >
            <div
              class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-dvqt9d-MuiInputBase-root-MuiFilledInput-root"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1gctnaj-MuiInputBase-input-MuiFilledInput-input"
                id=":ri:"
                placeholder=""
                role="combobox"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        class="MuiTypography-root MuiTypography-lightError css-1o6qin5-MuiTypography-root"
      />
    </div>
    <div>
      <div
        class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root"
      >
        <div
          class="MuiBox-root css-1i9vumk"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gobr0o-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root custom-input css-2vwk0i-MuiFormControl-root-MuiTextField-root"
            tabindex="0"
            withicon="true"
          >
            <div
              class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-dvqt9d-MuiInputBase-root-MuiFilledInput-root"
            >
              <input
                aria-autocomplete="list"
                aria-expanded="false"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1gctnaj-MuiInputBase-input-MuiFilledInput-input"
                id=":rk:"
                placeholder=""
                role="combobox"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        class="MuiTypography-root MuiTypography-lightError css-14nk251-MuiTypography-root"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-16awh2u-MuiAutocomplete-root"
    >
      <div
        class="MuiBox-root css-1i9vumk"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1gobr0o-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root custom-input css-2vwk0i-MuiFormControl-root-MuiTextField-root"
          tabindex="0"
          withicon="true"
        >
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-dvqt9d-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-autocomplete="list"
              aria-expanded="false"
              aria-invalid="false"
              autocapitalize="none"
              autocomplete="off"
              class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-1gctnaj-MuiInputBase-input-MuiFilledInput-input"
              id=":rk:"
              placeholder=""
              role="combobox"
              spellcheck="false"
              type="text"
              value=""
            />
            <div
              class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
            >
              <button
                aria-label="Open"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                tabindex="-1"
                title="Open"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      class="MuiTypography-root MuiTypography-lightError css-14nk251-MuiTypography-root"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
